pkg/detectors/aws/aws.go

```diff
@@ -136,9 +166,18 @@ func (s scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				},
 			}
 
+			account, err := GetAccountNumFromAWSID(resIDMatch)
```

Comment:
question: do we want to do this regardless of if the API call is successful or not, or did we only want to do this if the API call is unsuccessful? I think it works either way, I was just curious.

Reply:
I guess it'd be more performant if we only call this function when we do not return a successful API call from a valid credential.